src/Interpreters/Cache/FileCache.cpp

```diff
@@ -185,6 +187,20 @@ FileSegments FileCache::getImpl(
     /// Given range = [left, right] and non-overlapping ordered set of file segments,
     /// find list [segment1, ..., segmentN] of segments which intersect with given range.
 
+    FileSegments result;
+
+    if (enable_bypass_cache_with_threashold && (range.size() > bypass_cache_threashold))
```

Comment:
By the way, it is not the size of the file right? But just a range from file, and it's size depends on other settings?

Reply:
This is a range that is determined as:
`left bound` - the result of the seek (but it is not always the same as the result of last seek on the buffer, because AsynchronousReadBufferFromRemoteFS does seek avoiding). 
`right bound` - the result of `setReadUntilPosition` in `MergeTreeReaderStream::adjustRightMark` for wide parts, for compact parts there is a similar method.

Also see comment https://github.com/ClickHouse/ClickHouse/pull/42418#discussion_r998151807.